### PR TITLE
Deployment preparation: `ChildChainGaugeFactory`

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `ChildChainGaugeFactory`

Build info: master@https://github.com/balancer/balancer-v2-monorepo/commit/044d443737c2ad89ca0f3f2614298e4c911b3088

This fork test actually integrates everything (gauge factory, gauge, pseudo minter, VE proxy, VE boost, bridged VE -mocked in this PR-), so there is plenty of use cases to test. Input is welcome!

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

**Comment by jubeira** [comment]
Rebased on top of #2375 to use deployed addresses for VE proxy and L2 pseudo minter in the test (see https://github.com/balancer/balancer-v2-monorepo/pull/2372/commits/259e6bddfd224c3b39947e5eea0c65bddd43e2b8 for reference)

**Review by nventuro — APPROVED** [review]

File: pkg/deployments/tasks/20230316-child-chain-gauge-factory-v2/test/test.fork.ts [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+            { from: pseudoMinter.address, to: user2.address },
+            BAL.address
+          );
+
+          expect(user2Rewards.args.value).to.be.almostEqual(rewardUser2);
+        });
+
+        it('updates claimable tokens', async () => {
+          await pseudoMinter.connect(user1).mint(gauge.address);
+          await pseudoMinter.connect(user2).mint(gauge.address);
```
[the remark this answers is not in the document]
Comment (jubeira) [reply pkg/deployments/tasks/20230316-child-chain-gauge-factory-v2/test/test.fork.ts]: nice catch!

[the remark this answers is not in the document]
Comment (jubeira) [reply pkg/deployments/tasks/20230316-child-chain-gauge-factory-v2/test/test.fork.ts]: Addressed in https://github.com/balancer/balancer-v2-monorepo/pull/2372/commits/f91724d4079c1f13916083e79bc62d66ffbf5f8c

File: pkg/deployments/tasks/20230316-child-chain-gauge-factory-v2/test/contracts/IArbToken.sol [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * @notice should decrease token supply by amount, and should (probably) only be callable by the L1 bridge.
+     */
+    function bridgeBurn(address account, uint256 amount) external;
+
+    /**
+     * @return address of layer 1 token
+     */
+    function l1Address() external view returns (address);
+}
```
[the remark this answers is not in the document]
Comment (jubeira) [reply pkg/deployments/tasks/20230316-child-chain-gauge-factory-v2/test/contracts/IArbToken.sol]: Addressed in https://github.com/balancer/balancer-v2-monorepo/pull/2372/commits/f91724d4079c1f13916083e79bc62d66ffbf5f8c.

**Comment by jubeira** [comment]
Thanks for the review! I'll merge after CI finishes.

**Pull request merged**